app/src/main/java/com/example/faridaghadyali/myapplication/ListFragment.java

```diff
@@ -0,0 +1,174 @@
+                    //view to reflect the new item added to the list
+                }
+            }
+        });
+        addItemDialog.setNegativeButton("no", null);
+        addItemDialog.show();
+    }
+
+    public void createDeleteItemDialog(final int position){
+        //takes as input the position of the item that the user has long-clicked and creates an
+        //alert dialog when called. The user can click yes in the alert dialog to delete the item
+        //from the list or no to cancel deleting the item from the list
+        final String delete_item = (String) (listView.getItemAtPosition(position));
+        AlertDialog.Builder deleteItemDialog = new AlertDialog.Builder(getActivity());
+        deleteItemDialog.setTitle("Delete item from list");
+        deleteItemDialog.setPositiveButton("yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                //removes the selected item from the ArrayList
+                if (user_item.length() != 0) {
+                    items.remove(delete_item);
+                    itemsAdapter.notifyDataSetChanged(); //uses the ArrayAdapter to update the
+                    //view to reflect the item has been deleted from the list
+                }
+            }});
+        deleteItemDialog.setNegativeButton("no", null);
+        deleteItemDialog.show();
+    }
+
+    public void createEditItemDialog(final int position){
+        //takes as input the position of the item that the user has clicked and creates an alert
+        // dialog when called. The alert dialog has an input text field that contains the item the
+        //user clicked and the user can edit the item and click yes to apply the edit or no to
+        //cancel changing the item in the list
+        final String edit_item = (String) (listView.getItemAtPosition(position));
+        AlertDialog.Builder editItemDialog = new AlertDialog.Builder(getActivity());
+        editItemDialog.setTitle("Edit item");
+        final EditText input = new EditText(getActivity());
+        input.setText(edit_item);
+        editItemDialog.setView(input);
+        editItemDialog.setPositiveButton("yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                //sets the edited item to the position it previously occupied in the list
+                user_item = input.getText().toString().trim();
+                if (user_item.length() != 0) {
+                    items.set(position, user_item);
+                    itemsAdapter.notifyDataSetChanged(); //uses the ArrayAdapter to update the
+                    //view to reflect the item has been deleted from the list
+                }
+            }
+        });
+        editItemDialog.setNegativeButton("no", null);
+        editItemDialog.show();
+    }
+
+
+}
```

Comment:
You can basically have just one comment per method  that explains the whole thing- for future reference. Good job dividing stuff into methods.